internal/testutil/ibc/mock.go

```diff
@@ -10,34 +10,31 @@ import (
 )
 
 // BaseIBCMock returns a mock IBC module without a Host
-func BaseIBCMock(t gocuke.TestingT, busMock *mockModules.MockBus) *mockModules.MockIBCModule {
+func BaseIBCMock(t gocuke.TestingT, bus modules.Bus) *mockModules.MockIBCModule {
```

Comment:
Can you speak to your experience regarding the parameter type when passing a bus argument? I don't have an informed enough opinion yet and it seems like you've encountered something here.

Reply:
This is for use in the consensus e2e tests it uses a real bus not a mock so the arg must be a real one as this is where its being used currently

Reply:
I was looking for more about the "why" bus over mock bus, what are the constraints / requirements that drive your decision here?

Reply:
So if we look in `consensus/e2e_tests/utils_test.go` at the function `CreateTestConsensusPocketNode()` it is passed a real bus. As such it gives a real bus to each of the mocks it creates. I cannot use the real mock as a bus mock in this base function call and as its the only place using it, for now I thought its best to just follow the pattern and use the real bus its given.

In reality the IBC module is mostly a shepherd for the IBC host, I think any real mock that uses the IBC module will probably just need it to create the IBC host (or another submodule later down the line for servicers to handle IBC relaying). These submodules are registered to the bus, and are only retrievalbe from the bus - so a real bus makes sense here?